### PR TITLE
NUnit headless integration + XUnit test attributes 

### PR DESCRIPTION
**Pull request by maxkatz6**
## What does the pull request do?

Continues https://github.com/AvaloniaUI/Avalonia/pull/10473
Implements Headless NUnit integration, also does quite some changes with XUnit integration.

### Removed API:
```csharp
public class AvaloniaTestFrameworkAttribute(Type appBuilderEntryPointType) {}
```
### New APIs.
Avalonia.Headless (shared part, to make third parties to implement the same for MSTest and others):
```csharp
/// <summary>
/// Sets up global avalonia test framework using avalonia application builder passed as a parameter.
/// </summary>
public class AvaloniaTestApplicationAttribute(Type appBuilderEntryPointType) { }

public class HeadlessUnitTestSession
{
   static HeadlessUnitTestSession StartNew<TEntryPointType>();
   static HeadlessUnitTestSession StartNew(Type entryPointType);

    /// <summary>
    /// Creates a session from AvaloniaTestApplicationAttribute attribute or reuses any existing.
    /// If AvaloniaTestApplicationAttribute doesn't exist, empty application is used. 
    /// </summary>
   static HeadlessUnitTestSession GetOrStartForAssembly(Assembly? assembly);

   public Task<TResult> Dispatch<TResult>(Func<Task<TResult>> action, CancellationToken cancellationToken);

   public void Dispose();
 }
```
Avalonia.Headless.XUnit:

```csharp
public class AvaloniaTestFrameworkAttribute() {}
public class AvaloniaFactAttribute : FactAttribute {}
public class AvaloniaTheoryAttribute : TheoryAttribute {}
```

Avalonia.Headless.NUnit:

```csharp
public class AvaloniaTestAttribute : TestAttribute {}
public class AvaloniaTheoryAttribute : TheoryAttribute {}
```

## Checklist

- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [x] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Breaking changes

No, as previous XUnit package wasn't shipped to the nuget.

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0033864-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0033928-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0033974-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034018-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034106-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by maxkatz6** [comment]
@grokys @kekekeks ready to review as https://github.com/AvaloniaUI/AvaloniaEdit/pull/332 tests are now passing (with local builds at least). The most important part to review is HeadlessUnitTestSession, as it's shared between different unit test frameworks and is part of public API.
In short, on each test we now start a new Application instance with scoped Avalonia locator. In other words, the same how it works in our old unit tests. All Avalonia tests are executed on the single UI thread to keep it in a queue.

File: src/Headless/Avalonia.Headless.NUnit/AvaloniaTestMethodCommand.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+namespace Avalonia.Headless.NUnit;
+
+internal class AvaloniaTestMethodCommand : DelegatingTestCommand
+{
+    private readonly HeadlessUnitTestSession _session;
+    private readonly List<Action> _beforeTest;
+    private readonly List<Action> _afterTest;
+
+    private static FieldInfo s_innerCommand = typeof(DelegatingTestCommand)
+        .GetField("innerCommand", BindingFlags.Instance | BindingFlags.NonPublic)!;
```
Comment (grokys) [inline src/Headless/Avalonia.Headless.NUnit/AvaloniaTestMethodCommand.cs:19]: This is gnarly, but @maxkatz6 doesn't think there's currently a better way to do it with nunit.

**Review by grokys — APPROVED** [review]

File: src/Headless/Avalonia.Headless.NUnit/AvaloniaTestMethodCommand.cs (hunk 1 above)
Comment (maxkatz6) [reply src/Headless/Avalonia.Headless.NUnit/AvaloniaTestMethodCommand.cs:19]: Yes, will add a comment with details and links.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034498-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034508-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**